### ISSUE TITLE
The MouseEvent type on the YAxis is incorrect.

### ISSUE THREAD

**Issue by navyjeongs** [recharts/recharts#4959]
## Do you want to request a feature or report a bug?
YAxis MouseEvent has incorrect type

ex: `onClick`, `onMouseEnter`, `onMouseLeave`

## What is the current behavior?

`YAxis` onClick type is `onClick?: React.MouseEventHandler<SVGElement> | undefined`

The same occurs on the `XAxis`.

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar

```typescript
<ResponsiveContainer>
  <BarChart>
    {/* Target signature provides too few arguments. Expected 3 or more, but got 1 */}
    <YAxis onMouseEnter={(data, index, event) => {}} />


    <Bar onMouseEnter={(data, index, event) => {}} />
  </BarChart>
</ResponsiveContainer>
```

<img width="941" alt="스크린샷 2024-09-04 오후 5 14 33" src="https://github.com/user-attachments/assets/97394919-842f-482f-9988-f9935f4e528f">

## What is the expected behavior?

The `onClick` prop of the Bar component has the type  `AdaptChildMouseEventHandler<any, SVGPathElement> | undefined`

So, I think The `onClick` prop of the YAxis component has the type  `AdaptChildMouseEventHandler<any, SVGElement> | undefined`  or `(data: Payload, index: number, event: MouseEvent) => void;`

`Legend` component was fixed some time ago.
https://github.com/recharts/recharts/issues/2667

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?

OS/Browser: Mac/Chrome

recharts ver : 2.12.7

**Comment by ckifer** [comment]
PR up to fix this in 2.x. Type changes are generally breaking but in this case we're fixing something that is misleading

**Comment by ckifer** [comment]
Fixed in alpha https://www.npmjs.com/package/recharts/v/2.13.0-alpha.5

**Comment by ckifer** [comment]
Released in 2.13
